source/concurrent/Promise.ooc

```diff
@@ -109,7 +109,7 @@ Future: abstract class <T> {
 }
 
 _ThreadFuture: class <T> extends Future<T> {
-	_result: Object
+	_result: __onheap__ T
```

Comment:
And just when I thought ooc code can't get more confusing than it already is... Is inline assembly the next step ? :P